sbin/common/common.sh

```diff
@@ -173,7 +173,25 @@ function setBootJdk() {
     fi
 
     echo "Guessing JDK_BOOT_DIR: ${BUILD_CONFIG[JDK_BOOT_DIR]}"
-    echo "If this is incorrect explicitly configure JDK_BOOT_DIR"
+    echo "If this is incorrect explicitly configure JDK_BOOT_DIR using --jdk-boot-dir"
+
+    # Calculate version number of boot jdk
+    export BOOT_JDK_VERSION_NUMBER=$(${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -XshowSettings:properties -version 2>&1 | awk '/java.specification.version/{gsub(/1\./,"",$3);print $3}')
+    if [ -z "$BOOT_JDK_VERSION_NUMBER" ]; then
+      echo "[ERROR] The BOOT_JDK_VERSION_NUMBER was not found. Likelihood is that the boot jdk settings properties don't contain a java.specification.version..."
+      ${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -XshowSettings:properties -version 2>&1
+      exit 2
+    fi
+
+    # If bootjdk isn't the same version and isn't a n-1 version, crash out with informational message
+    export REQUIRED_BOOT_JDK=$((${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}-1))
+
+    if [ ${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]} -ne $BOOT_JDK_VERSION_NUMBER ] && [ $REQUIRED_BOOT_JDK -ne $BOOT_JDK_VERSION_NUMBER ]; then
```

Comment:
My bash on Windows complains about an integer expression expected.

Reply:
What's the full complaint message?

Reply:
`: integer expression expectedn/common/common.sh: line 189: [: 8`

You probably need double square brackets: `[[`.